HIPs/firefly_graduation.md

```diff
@@ -0,0 +1,177 @@
+# Hyperledger FireFly Graduation Proposal
+
+## Abstract
+
+FireFly began as a [Hyperledger Lab in June, 2021](https://wiki.hyperledger.org/display/TSC/Table+consideration+of+the+FireFly+proposal+for+two+months) and became an [Incubating Project in September, 2021](https://wiki.hyperledger.org/display/TSC/Move+Hyperledger+Lab+Firefly+into+a+Project+in+Incubation). Now, after two years in Incubation FireFly has significantly grown and continued to mature. The project has received significant contributions from committers beyond the set of original maintainers, including a completely new [blockchain connector microservice](https://github.com/hyperledger/firefly-tezosconnect) and corresponding FireFly Core plugin enabling connectivity to Tezos, an entirely new type of blockchain. Additionally, the project now has a diverse set of maintainers representing three different companies.
+
+Given that FireFly now meets or exceeds the [Incubation Exit Criteria](https://toc.hyperledger.org/governing-documents/project-incubation-exit.html) created by the Hyperledger TOC, is a proposal to move FireFly to Graduated Project status. The rest of this document describes in detail how the project meets or exceeds the Incubation Exit Criteria.
+
+## Minimum requirements
+
+### Legal
+
+> All code has been made available under the Apache License and is free of incompatible dependencies
+> Project name has been checked for trademark issues
+
+Both of these requirements were met before FireFly moved in the Hyperledger Labs.
+
+### Community support
+
+> The project must have an active and diverse set of contributing members representing various constituencies
+
+FireFly has a very active Discord channel where community members from all over the world regularly chat about project ideas, suggest improvements, ask technical questions, or seek troubleshooting assistance.
+
+The community holds public Community Calls at least once a month where maintainers and contributors discuss architectural proposals, upcoming releases, and demo new features in the project.
+
+> The project is not highly dependent on any single contributor (there are at least 3 legally independent committers and there is no single company or entity that is vital to the success of the project)
+
+#### Committers
+
+According to [LFX Insights](https://insights-v2.lfx.linuxfoundation.org/firefly/technical-contributors/participating-organizations?selectedDateFilterType=DATERANGE&selectedDateRangeKey=3Y), FireFly has committers representing 8 corporations and 12 independent contributors.
+
+#### Maintainers
+
+FireFly has 12 maintainers representing three different companies:
```

Comment:
With 10 maintainers coming from Kaleido, does this meet the `there is no single company or entity that is vital to the success of the project`? In other words, if Kaleido decided to stop supporting FireFly development, would the project continue to succeed? What is being done to move to a more diverse set of maintainers?

Reply:
> What is being done to move to a more diverse set of maintainers?

This is a good question. I'm happy to elaborate on this more in the TOC call tomorrow.

> Specifically, the non-Kaleido maintainers appear to be active in the connectors, have they started to maintain the core Firefly repositories as well?

Here are @denisandreenko's commits in FireFly Core: https://github.com/hyperledger/firefly/commits?author=denisandreenko (Many of these were made earlier in the year but the branch was rebased before it got merged.)

I will also make the case that contributions to connectors should not be considered "second class" to contributions to the main FireFly repo. We call the main repo "FireFly Core" because it is the piece that all other pieces connect to. But the "Core" without connectors is useless. Additionally, the connectors contain the some of the most difficult, blockchain-specific code in the project.